### PR TITLE
Added function-body-length-rule

### LINKED ISSUE #2881: Rule to count lines of methods, functions or arrow functions
### Bug Report

- __TSLint version__: 5.4.2
- __TypeScript version__: 2.3.0
- __Running TSLint via__: Visual Studio 

#### TypeScript code being linted

```ts
// code snippet
```

with `tslint.json` configuration:

```json

```

### LINKED ISSUE #3549: Port tslint-microsoft-contrib's max-func-body-length
See https://github.com/palantir/tslint/issues/1142.

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
#### PR checklist

- [x] Addresses an existing issue: Fixes #3549. Fixes #2881. Dup of #2882
- [x] New feature, bugfix, or enhancement
  - [x] Includes tests
- [x] Documentation update

#### Overview of change:

I removed shorthand names like 'func' and wrote everything out to be a bit more clear.

#### CHANGELOG.md entry:

[new-rule] function-body-length

### CODE REVIEW

File: src/configs/all.ts [hunk 1]
```diff
@@ -29,19 +29,25 @@ export const rules = {
     "ban-types": {
```
Comment (JoshuaKGoldberg) [inline src/configs/all.ts]: Agh

Comment (JoshuaKGoldberg) [reply src/configs/all.ts]: Agh, the Prettier changes...

**Comment by goktugyil** [comment]
why stopped working on this?

**Comment by JoshuaKGoldberg** [comment]
@goktugyil just waiting for the rest of the PR queue to be reviewed 😊 I have a few waiting.

**Comment by JoshuaKGoldberg** [comment]
I renamed it to `function-body-length` for a slightly shorter name.

File: tslint.json [hunk 2]
```diff
@@ -1,73 +1,64 @@
...
+        "switch-default": false,
+        "variable-name": [
+            true,
+            "ban-keywords",
+            "check-format",
+            "allow-leading-underscore",
+            "allow-pascal-case"
+        ],
+        "linebreak-style": false
+    }
```
Comment (goktugyil) [inline tslint.json]: Please try not to include non feature related changes in a PR since it is the top reason for a review to take longer than it needs to. Examples are: prettier, making stuff clearer, fixing typos, etc..

File: src/configs/recommended.ts [hunk 3]
```diff
@@ -32,9 +32,7 @@ export const rules = {
                 "Function",
                 "Avoid using the `Function` type. Prefer a specific function type, like `() => void`.",
             ],
-            ["Boolean", "Avoid using the `Boolean` type. Did you mean `boolean`?"],
             ["Number", "Avoid using the `Number` type. Did you mean `number`?"],
-            ["String", "Avoid using the `String` type. Did you mean `string`?"],
```
Comment (goktugyil) [inline src/configs/recommended.ts]: Why did you delete these?

File: test/rules/function-body-length/default/test.ts.lint [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+        1,
+~~~~~~~~~~
+        1,
+~~~~~~~~~~
+        1,
+~~~~~~~~~~
+    ];
+~~~~~~
+};
+~ [Maximum function expression body length exceeded: 101 lines instead of <= 100.]
```
Comment (goktugyil) [inline test/rules/function-body-length/default/test.ts.lint]: Surely there must be a better way to do this.

File: tslint.json (hunk 2 above)
Comment (JoshuaKGoldberg) [reply tslint.json]: @goktugyil thanks for stepping in to review these 😄 I appreciate the effort! I'm waiting on this for a review from another maintainer. This PR was sent back before Prettier was finished being integrated into the repo, so there are likely a bunch of unintended merge conflicts.

We should also note that this repo is being deprecated soon after wavering amounts of maintenance activity (#4534), so it's not likely we'll have time to get through a lot of these older PRs.

File: test/rules/function-body-length/default/test.ts.lint (hunk 4 above)
Comment (JoshuaKGoldberg) [reply test/rules/function-body-length/default/test.ts.lint]: I actually quite like the unit testing system here 😊

Comment (goktugyil) [reply test/rules/function-body-length/default/test.ts.lint]: 300 lines of `1`s 😁

**Comment by adidahiya** [comment]
Closing for housekeeping due to low priority. This rule implementation can live in tslint-microsoft-contrib if you need it
